pkg/providers/azure.go

```diff
@@ -193,7 +193,7 @@ func (sc *SetUpCmd) getTenantId(ctx context.Context) error {
 	if len(tenants) > 1 {
 		return errors.New("multiple tenants found")
 	}
-	sc.tenantId = fmt.Sprint(&tenants[0])
+	sc.tenantId = *tenants[0].TenantID
```

Comment:
I know the sdk's can be difficult to test, but might be something we look into options for

Reply:
Agreed we can at least mock the calls to the api